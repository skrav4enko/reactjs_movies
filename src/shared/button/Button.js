import React from 'react';
import './Button.scss';

function Button(props) {
  const { children, type = 'button', variant, size, onClick } = props;

  return (
    <button className={`button ${variant} ${size}`} type={type} onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
