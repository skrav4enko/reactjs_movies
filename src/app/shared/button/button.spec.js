/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('Button', () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      children: 'Name',
      type: 'button',
      variant: 'primary',
      size: 'small',
      onClick: jest.fn(),
    };
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<Button {...defaultProps}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default type button', () => {
    const testProps = {
      ...defaultProps,
      type: undefined,
    }
    const wrapper = shallow(<Button {...testProps}/>);
    expect(wrapper.prop('type')).toEqual('button');
  });
});
