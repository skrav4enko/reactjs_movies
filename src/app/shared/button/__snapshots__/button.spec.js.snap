// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="button primary small"
  onClick={[MockFunction]}
  type="button"
>
  Name
</button>
`;
